internal/cortex/querier/queryrange/instrumentation.go

```diff
@@ -12,19 +12,51 @@ import (
 	"github.com/weaveworks/common/instrument"
 )
 
+const DAY = 24 * time.Hour
+const QUERYDURATIONBUCKET = "query_range_bucket"
+
 // InstrumentMiddleware can be inserted into the middleware chain to expose timing information.
 func InstrumentMiddleware(name string, metrics *InstrumentMiddlewareMetrics) Middleware {
-	var durationCol instrument.Collector
-
-	// Support the case metrics shouldn't be tracked (ie. unit tests).
-	if metrics != nil {
-		durationCol = instrument.NewHistogramCollector(metrics.duration)
-	} else {
-		durationCol = &NoopCollector{}
-	}
 
 	return MiddlewareFunc(func(next Handler) Handler {
 		return HandlerFunc(func(ctx context.Context, req Request) (Response, error) {
+
+			var durationCol instrument.Collector
+			// Support the case metrics shouldn't be tracked (ie. unit tests).
+			if metrics != nil {
+				durationCol = instrument.NewHistogramCollector(metrics.duration)
+			} else {
+				durationCol = &NoopCollector{}
+			}
+			queryRangeDuration := req.GetEnd() - req.GetStart()
+			var queryRangeDurationBucket string
+			switch {
+			case queryRangeDuration < 0:
+				queryRangeDurationBucket = "Invalid"
+			case queryRangeDuration >= 0 && queryRangeDuration <= time.Hour.Milliseconds():
+				queryRangeDurationBucket = "1h"
+			case queryRangeDuration > time.Hour.Milliseconds() &&
+				queryRangeDuration <= 6*time.Hour.Milliseconds():
+				queryRangeDurationBucket = "6h"
+			case queryRangeDuration > 6*time.Hour.Milliseconds() &&
+				queryRangeDuration <= 12*time.Hour.Milliseconds():
+				queryRangeDurationBucket = "12h"
+			case queryRangeDuration > 12*time.Hour.Milliseconds() &&
+				queryRangeDuration <= DAY.Milliseconds():
+				queryRangeDurationBucket = "1d"
+			case queryRangeDuration > DAY.Milliseconds() &&
+				queryRangeDuration <= 2*DAY.Milliseconds():
+				queryRangeDurationBucket = "2d"
+			case queryRangeDuration > 2*DAY.Milliseconds() &&
+				queryRangeDuration <= 7*DAY.Milliseconds():
+				queryRangeDurationBucket = "7d"
+			case queryRangeDuration > 7*DAY.Milliseconds() &&
+				queryRangeDuration <= 30*DAY.Milliseconds():
+				queryRangeDurationBucket = "30d"
+			case queryRangeDuration > 30*DAY.Milliseconds():
```

Comment:
Looks like you are using switch case to do if else, or use `default`

Reply:
sg